specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -0,0 +1,158 @@
+
+### Overwriting Metadata
+Automatically capturing cloud metadata doesn't work reliably from a Lambda environment. Moreover, retrieving cloud metadata through an additional HTTP request may slowdown the lambda function / increase cold start behaviour. Therefore, the generic cloud metadata fetching should be disabled when the agent is running in a lambda context (for instance through checking for the existance of the `AWS_LAMBDA_FUNCTION_NAME` environment variable). For AWS Lambda we adapt meta-data fetching that uses available environment variables to derive / overwrite the following fields:
+
+Field | Value | Description 
+---   | ---   | ---
+`service.name` | e.g. `MyFunctionName` | The name of the Lambda function. Retrieved from the environment variable `AWS_LAMBDA_FUNCTION_NAME`.
+`service.framework.name` | `AWS Lambda` | Contant value for the framework name.
+`service.runtime.name`| e.g. `AWS_Lambda_java8` | The lambda runtime derived from the `AWS_EXECUTION_ENV` environment variable.
+`cloud.provider` | `aws` | Contant value for the cloud provider.
+`cloud.region` | e.g. `us-east-1` | The cloud region derived from the `AWS_REGION` environment variable.
+`cloud.service.name` | `lambda` |  The AWS service which is the value `lambda` for this instrumentation. 
+`faas.name` | e.g. `MyFunctionName` | The lambda function name derived from the `AWS_LAMBDA_FUNCTION_NAME` environment variable.
+`faas.version`| e.g. `${LATEST}` | The lambda function version derived from the `AWS_LAMBDA_FUNCTION_VERSION` environment variable.
+
+
+### Deriving cold starts 
+A cold start occurs if AWS needs first to initialize the Lambda runtime (including the Lambda process, such as JVM, Node.js process, etc.) in oder to handle a request. This happens for the first request and after long function idle times. Lambda functions are always called sequentially (there is no concurrency). Thus, detecting a cold start is as simple as detecting whether the invocation of a __handler method__ is the **first since process startup** or not. This can be achieved with a global / process-scoped flag that is flipped at the first execution of the handler method.
+
+
+
+## Trigger-specific Instrumentation
+Lambda functions can be triggered in many different ways. A generic transaction for a Lambda invocation can be create independently of the actual trigger. However, depending on the trigger type different information might be available that can be used to capture additional transaction data or allows to derive additional, valuable spans. The most common triggers that we want dedicated instrumentation support for are the following:  
+
+- API Gateway V1 
+- API Gateway V2 
+- SQS
+- SNS
+- S3
+
+If none of the above apply, the fallback should be a generic instrumentation (as described above) that can deal with any type of trigger (thus capturing only the minimal available information).
+
+### API Gateway (V1 & V2)
```

Comment:
For API Gateway triggers, in the python agent we are forming the transaction name as `$http_method $function_name` (e.g. `GET my_handler`), as this matches incoming transactions in a common web framework. I think this is a pattern worth adopting and we should add it to the spec here.

Reply:
agree, will adapt the spec.